src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseExportService.java

```diff
@@ -149,6 +157,115 @@ private void deleteTempZipRepoFiles(List<Path> pathsToZipeedRepos) {
+     * @throws IOException is thrown for file handling errors
+     */
+    public File checkPlagiarism(long programmingExerciseId) throws ExitException, IOException {
+        ProgrammingExercise programmingExercise = programmingExerciseRepository.findWithAllParticipationsById(programmingExerciseId).get();
+
+        programmingExercise.getStudentParticipations().parallelStream().forEach(participation -> {
+            var programmingExerciseParticipation = (ProgrammingExerciseParticipation) participation;
+            try {
+                if (programmingExerciseParticipation.getRepositoryUrlAsUrl() == null) {
+                    log.warn("Ignore participation " + participation.getId() + " for export, because its repository URL is null");
+                    return;
+                }
+                Repository repo = gitService.getOrCheckoutRepository(programmingExerciseParticipation, REPO_DOWNLOAD_CLONE_PATH);
+                gitService.resetToOriginMaster(repo); // start with clean state
+
+                // TODO: offer the following options in the client
+                // 1) filter empty submissions, i.e. repositories with no student commits
+                // 2) filter submissions with a result score of 0%
+
+                repo.close();
+            }
+            catch (GitException | GitAPIException | InterruptedException ex) {
+                log.error("clone student repository " + programmingExerciseParticipation.getRepositoryUrlAsUrl() + " in exercise '" + programmingExercise.getTitle()
+                        + "' did not work as expected: " + ex.getMessage());
+            }
+        });
+
+        var output = "output";
+        var templateRepoName = "";
+        // clone the template repo
+        try {
+            Repository templateRepo = gitService.getOrCheckoutRepository(programmingExercise.getTemplateParticipation(), REPO_DOWNLOAD_CLONE_PATH);
+            templateRepoName = versionControlService.get().getRepositorySlugFromUrl(programmingExercise.getTemplateParticipation().getRepositoryUrlAsUrl());
+            gitService.resetToOriginMaster(templateRepo); // start with clean state
+            templateRepo.close();
+        }
+        catch (GitException | GitAPIException | InterruptedException ex) {
+            log.error("clone template repository " + programmingExercise.getTemplateParticipation().getRepositoryUrlAsUrl() + " in exercise '" + programmingExercise.getTitle()
+                    + "' did not work as expected: " + ex.getMessage());
+        }
+
+        var projectKey = programmingExercise.getProjectKey();
+        var outputFolder = REPO_DOWNLOAD_CLONE_PATH + (REPO_DOWNLOAD_CLONE_PATH.endsWith(File.separator) ? "" : File.separator) + projectKey + "-" + output;
+
+        File outputFolderFile = new File(outputFolder);
+        outputFolderFile.mkdirs();
+
+        var programmingLanguage = "";
+        switch (programmingExercise.getProgrammingLanguage()) {
```

Comment:
Codacy found an issue: [Switch statements should have a default label](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968852)